src/main/java/eu/dissco/nusearch/configuration/RabbitMQConfiguration.java

```diff
@@ -0,0 +1,40 @@
+package eu.dissco.nusearch.configuration;
+
+import eu.dissco.nusearch.component.MessageCompressionComponent;
+import eu.dissco.nusearch.property.RabbitMQProperties;
+import lombok.AllArgsConstructor;
+import org.springframework.amqp.rabbit.config.SimpleRabbitListenerContainerFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@AllArgsConstructor
+public class RabbitMQConfiguration {
+
+  private final MessageCompressionComponent compressedMessageConverter;
+  private final RabbitMQProperties rabbitMQProperties;
+
+  @Bean
+  public SimpleRabbitListenerContainerFactory consumerBatchContainerFactory(
+      ConnectionFactory connectionFactory) {
```

Comment:
Where do you declare the ConnectionFactory bean?

Reply:
https://docs.spring.io/spring-framework/reference/core/beans/java/bean-annotation.html#beans-java-dependencies